File: 2523-closest-prime-numbers-in-range/2523-closest-prime-numbers-in-range.py
class Solution(object):
    def closestPrimes(self, left, right):
        limit = 10**6
        is_prime = [True] * (limit + 1)
        is_prime[0] = is_prime[1] = False 
        for i in range(2, int(limit**0.5) + 1):
            if is_prime[i]:
                for j in range(i * i, limit + 1, i):
                    is_prime[j] = False
        primes = [i for i in range(max(2, left), right + 1) if is_prime[i]]
        if len(primes) < 2:
            return [-1, -1]
        min_diff = float('inf')
        result = [-1, -1]
        for i in range(1, len(primes)):
            diff = primes[i] - primes[i - 1]
            if diff < min_diff:
                min_diff = diff
                result = [primes[i - 1], primes[i]]
        return result
            

File: 2523-closest-prime-numbers-in-range/2523-closest-prime-numbers-in-range.java
class Solution {
    public int[] closestPrimes(int left, int right) {
        int limit = 1000000;
        boolean[] isPrime = new boolean[limit + 1];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;
        for (int i = 2; i * i <= limit; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= limit; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        List<Integer> primes = new ArrayList<>();
        for (int i = Math.max(2, left); i <= right; i++) {
            if (isPrime[i]) {
                primes.add(i);
            }
        }
        if (primes.size() < 2) return new int[]{-1, -1};
        int minDiff = Integer.MAX_VALUE;
        int[] result = new int[2];
        for (int i = 1; i < primes.size(); i++) {
            int diff = primes.get(i) - primes.get(i - 1);
            if (diff < minDiff) {
                minDiff = diff;
                result[0] = primes.get(i - 1);
                result[1] = primes.get(i);
            }
        }

        return result;
    }
}